ui/admin/app/components/agent/shared/ToolAuthenticationStatus.tsx

```diff
@@ -56,7 +52,12 @@ export function ToolAuthenticationStatus({
 
 		if (error) return;
 
-		onUpdate({ ...toolInfo, authorized: false });
+		setToolInfo({
```

Comment:
When tool is being authorized or deauthorized, we should just remove the tool from toolInfo and repoll the agent until toolInfo is set.